api/src/main/java/org/apache/iceberg/actions/DeleteOrphanFiles.java

```diff
@@ -80,9 +84,16 @@ public interface DeleteOrphanFiles extends Action<DeleteOrphanFiles, DeleteOrpha
    *
    * @param executorService the service to use
    * @return this for method chaining
+   * @deprecated All deletes should be performed using the bulk delete api if available. Use FileIO
+   *     specific parallelism controls to adjust bulk delete concurrency within that api.
    */
+  @Deprecated
   DeleteOrphanFiles executeDeleteWith(ExecutorService executorService);
 
+  default DeleteOrphanFiles deleteBulkWith(Consumer<Iterable<String>> deleteFunc) {
```

Comment:
Do we have a use case for this? If not, I'd skip it and just support the old `deleteWith` method.

Reply:
I'd rather remove both, but this is required for testing the bulk delete method. The other option is I can rewrite the test suites which rely on "deleteWith" to instead just delete files for real.

Reply:
My worry is that the current implementation breaks the current behavior in an incompatible way: it checks if the table IO supports bulk deletes and if yes, ignores `deleteWith`. It will break those dry-run scenarios if the table is configured with a non-bulk IO and the user passes a custom delete function.

Can you elaborate a bit on why this is needed? Why not just not go through the bulk delete API if a custom delete function is provided?

Reply:
My goal is to remove the non-bulk delete api completely in the future. I really don't want to be in a situation where we have some apis that only work in conjunction with others.

I think if we allow `executeDeleteWith` and `deleteWith` we essentially have this permanently split api. Where sometimes these functions are able to do something but most of the time they can not. In the future I want to be in a situation where there is only `deleteBulkWith`.

If we don't add `deleteBulkWith` but keep an alternate pathway with `executeDeleteWith` and `deleteBulkWith` we will permanently need two sets of tests. One testing the bulk pathway and one testing the non-bulk pathway. Which pathway is used will also not be super clear as some procedures (if we keep dry run as is) will invoke the non bulk pathway.